components/renku_data_services/secrets/db.py

```diff
@@ -108,3 +111,50 @@ async def delete_secret(self, requested_by: APIUser, secret_id: str) -> None:
                 return None
 
             await session.execute(delete(SecretORM).where(SecretORM.id == secret.id))
+
+    async def get_all_secrets_batched(
+        self, batch_size: int = 100
+    ) -> AsyncGenerator[Sequence[tuple[Secret, str]], None]:
+        """Get secrets in batches.
+
+        Only for internal use.
```

Comment:
I wanted to originally but thought it weird to have a function argument that the function doesn't actually use. But since you mention having a different id type, I think that could work, at least then the function can check that `id == "secrets_rotation"` so if would actually use the parameter. So I'll change that.